frappe/database/mariadb/database.py

```diff
@@ -236,7 +236,12 @@ def is_type_datetime(code):
 	def rename_table(self, old_name: str, new_name: str) -> list | tuple:
 		old_name = get_table_name(old_name)
 		new_name = get_table_name(new_name)
-		return self.sql(f"RENAME TABLE `{old_name}` TO `{new_name}`")
+
+		if new_name in self.get_tables(cached=False) and frappe.flags.in_migrate:
+			print(f"Rename Table: Dropping existing table with same name {new_name}")
+			self.sql_ddl(f"DROP TABLE IF EXISTS `{new_name}`")
```

Comment:
scary.

We should show a log summary of DDL changes made in each migration ig

Reply:
This only happens for explicit `rename_doctypes` in patch. So I think it's fine. 


The problem occurs when a patch that renames doctype fails and needs to be reattempted. There's no other option but to drop old table or skip the patch.